saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/web/Saml2MetadataFilterTests.java

```diff
@@ -120,4 +121,17 @@ public void setRequestMatcherWhenNullThenIllegalArgument() {
 		assertThatIllegalArgumentException().isThrownBy(() -> this.filter.setRequestMatcher(null));
 	}
 
+	@Test
```

Comment:
Can you add a test that shows that when the metadata filename is set, then it's used when the filter is invoked?